cargo-pgx/src/templates/cargo_config

```diff
@@ -1,3 +1,13 @@
-[build]
-# Postgres symbols won't ve available until runtime
-rustflags = ["-C", "link-args=-Wl,-undefined,dynamic_lookup"]
+# Auto-generated by pgx. You may edit this, or delete it to have a new one created.
```

Comment:
I would love this too!

According to https://doc.rust-lang.org/cargo/reference/config.html#target we can only set the `linker` value in `[target.'<triple>']`, unfortunately, `[target.'cfg(target_arch = "x86_64")']` and other expressions only allow setting `runner` and `rustflags`.

```toml
[target.'cfg(target_arch = "x86_64")']
linker = "./.cargo/pgx-linker-script.sh"

# results in:
# warning: unused key `linker` in [target] config table `cfg(target_arch = "x86_64")`
```

Interestingly, this doesn't return a warning but also doesn't work:

```toml
[target."*"]
linker = "./.cargo/pgx-linker-script.sh"
```

The primary design problem here is we need to access the environment variable `$CARGO_MANIFEST_DIR` or otherwise somehow reference a relative directory in something like `build.rustflags` of the `.cargo/config`. The only way I've found to do that is via a target triple specific setting. :(